File: backend/src/main/java/ru/bowling/bowlingapp/DTO/BowlingClubDTO.java
package ru.bowling.bowlingapp.DTO;

import jakarta.validation.constraints.*;
import lombok.*;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BowlingClubDTO {
    @NotBlank
    private String name;

    @NotBlank
    private String address;

    @NotNull
    @Min(1)
    private Integer lanesCount;

    private String contactPhone;
    private String contactEmail;
    
    // Типы оборудования в клубе
    private List<ClubEquipmentTypeDTO> equipmentTypes;
}

File: frontend/lib/models/bowling_club_dto.dart
class BowlingClubDto {
  final String name;
  final String address;
  final int lanesCount;
  final String? contactPhone;
  final String? contactEmail;

  BowlingClubDto({
    required this.name,
    required this.address,
    required this.lanesCount,
    this.contactPhone,
    this.contactEmail,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'address': address,
        'lanesCount': lanesCount,
        'contactPhone': contactPhone,
        'contactEmail': contactEmail,
      };

  factory BowlingClubDto.fromJson(Map<String, dynamic> json) {
    return BowlingClubDto(
      name: json['name'] as String,
      address: json['address'] as String,
      lanesCount: json['lanesCount'] is int
          ? json['lanesCount'] as int
          : int.tryParse(json['lanesCount']?.toString() ?? '0') ?? 0,
      contactPhone: json['contactPhone'] as String?,
      contactEmail: json['contactEmail'] as String?,
    );
  }
}
